backend/DevUp/DevUp.Repository/Repositories/IRepository.cs

```diff
@@ -0,0 +1,13 @@
+﻿using DevUp.Domain.Entities;
+
+namespace DevUp.Repository.Repositories
+{
+    public interface IRepository<T, T1> where T : class
+    {
+        Task<T> GetByIdAsync(IEntityId<T1> id);
+        Task<IReadOnlyList<T>> GetAllAsync();
+        Task<T> AddAsync(T entity);
+        Task<T> UpdateAsync(T entity);
+        Task<T1> DeleteAsync(IEntityId<T1> id);
+    }
```

Comment:
I liked previous approach with `TId` instead of `T1` better. It's easier to read. We should stay consistent.

Also, it feels like `T` should be more constrained. I feel like we should have two hierarchies: one for entities, one for their ids. Repository-wise, it'd look like:
`public interface IRepository<TEntity, TEntityId> where TEntity : Entity<TEntityId>`

This would allow to encapsulate shared behavior for entities in their base `abstract class Entity`. All of them definitely need an ability to be compared with other entities, so that's one for sure.